### ISSUE TITLE
qemu-xen doesn't support read-only disk drivers

### ISSUE THREAD

**Issue by gbirincsik** [OpenNebula/addon-xen#13]
Hey Guys,

When I try to deploy a Xen HVM VM I got the following message:

Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: libxl: error: libxl_dm.c:717:libxl__build_device_model_args_new: qemu-xen doesn't support read-only disk drivers
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: libxl: error: libxl_dm.c:1393:device_model_spawn_outcome: (null): spawn failed (rc=-3)
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: libxl: error: libxl_create.c:1188:domcreate_devmodel_started: device model did not start: -3
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: libxl: error: libxl_dm.c:1489:kill_device_model: unable to find device model pid in /local/domain/100/image/device-model-pid
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: libxl: error: libxl.c:1437:libxl__destroy_domid: libxl__destroy_device_model failed for 100
Wed Dec  7 11:51:29 2016 [Z0][VMM][E]: Unable
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: ExitCode: 3
Wed Dec  7 11:51:29 2016 [Z0][VMM][I]: Failed to execute virtualization driver operation: deploy.
Wed Dec  7 11:51:29 2016 [Z0][VMM][E]: Error deploying virtual machine: Unable
Wed Dec  7 11:51:29 2016 [Z0][VM][I]: New LCM state is BOOT_FAILURE

Any idea how I can fix it?

**Comment by JOJ0** [comment]
this is probably because xen tries to use the a disk using the IDE driver (/dev/hd*).
i had this problem and it was because of the context iso being put into as hd*
in oned.conf you can configure that it is also using sd* for cdrom disks.

**Comment by gbirincsik** [comment]
Lemme try it, I'll come back shortly.

**Comment by JOJ0** [comment]
it's this option in oned.conf:

DEFAULT_CDROM_DEVICE_PREFIX = "sd"

**Comment by gbirincsik** [comment]
Same error :(  I assume I did something incorrectly. May I ask a private contact, and we can discuss it for example via skype for example.

**Comment by JOJ0** [comment]
no, better discuss this in the opennebula forum, so others could also learn from it!

**Comment by gbirincsik** [comment]
Here the generated config:

name = 'one-43'
#O CPU_CREDITS = 512
memory  = '2048'
builder = 'hvm'
disk = [
    'raw:/var/lib/one//datastores/0/43/disk.0,sdc,w',
    'raw:/var/lib/one//datastores/0/43/disk.1,sda,w',
    'raw:/var/lib/one//datastores/0/43/disk.2,sdb,r',
]
vif = [
    ' mac=02:00:0a:01:00:dc,ip=10.1.0.220,bridge=mgmt',
]
vnc = '1'
vnclisten = '0.0.0.0'
vncunused = '0'
vncdisplay = '43'

I get same error message with it.

**Comment by JOJ0** [comment]
c'mon make a forum post please ;-) i dont think it's an issue in the code so we shouldn't discuss here, but a configuration error. i had this and fixed this but it's some time ago, not sure it it's the same.
i also i did some forum posts about it...

**Comment by jmelis** [comment]
Hey @JOJ0, thanks for encouraging the forum use :)

**Comment by gbirincsik** [comment]
I opened a topic about this on the opennebula forum as well.

**Comment by JOJ0** [comment]
ah ok Gabor, I didn't check the forum, sorry. please post the config you just posted here also in the forum

**Comment by gbirincsik** [comment]
I have copied just now.
